### ISSUE TITLE
Abductor's omnitool has a few tools that doesn't work

### ISSUE THREAD

**Issue by Singul0** [tgstation/tgstation#82439]
## Issue Summary
Tool options in the abductor omnitools such as (but not limited to): blood_filters, welding_tool

## Round ID:
N/A

## Testmerges:
N/A

## Reproduction:
1. spawn abductor omnitool
2. test the more exotic tools such as bloodfilter and welding tool
3. doesn't work

**Comment by 00-Steven** [comment]
oh boy

**Comment by 00-Steven** [comment]
yeah lemme take a look at that tomorrow

**Comment by Singul0** [comment]
Yeah, I was wondering how tf you managed implement all the tools without any being left out after I discovered you had a similar implementation to what im doing. Since I found out sone of the code isn't uniformly modularized

**Comment by 00-Steven** [comment]
Right, so, first thoughts are that they're somehow specialcased in some way, such that just having the tool type assigned isn't enough.
Which, well, looking at an example of welding code:
https://github.com/tgstation/tgstation/blob/03e466b15c87cbe83003d999149a6d32487bacbf/code/game/machinery/doors/airlock.dm#L1076-L1088
It doesn't _look_ like it, buuuuuuuuut, looks can be deceiving.
https://github.com/tgstation/tgstation/blob/03e466b15c87cbe83003d999149a6d32487bacbf/code/game/machinery/doors/airlock.dm#L1077
This line in particular seems like it may be related, as it seems like it's not _stopping_ welding actions but just _wordlessly failing_.
Looking at what this line is, we get:
https://github.com/tgstation/tgstation/blob/03e466b15c87cbe83003d999149a6d32487bacbf/code/game/objects/items.dm#L1119-L1127
Where, at base, the proc fails if amount is anything but 0.
https://github.com/tgstation/tgstation/blob/03e466b15c87cbe83003d999149a6d32487bacbf/code/game/objects/items/tools/weldingtool.dm#L293-L303
And hey, would you look at that, welding tools have their own implementation for this!
Buuuuut our omnitool doesn't seem to have that.
https://github.com/tgstation/tgstation/blob/66b4b0cde2d45d6829b4bf28f918421444cce789/code/modules/antagonists/abductor/equipment/gear/abductor_items.dm#L614-L732
 So, presumably, it's using the default implementation, and thus wordlessly failing the check.

**Comment by 00-Steven** [comment]
For blood filters, well, it also just changes the tool behaviour. But, just looking at the blood filter surgery step:
https://github.com/tgstation/tgstation/blob/66b4b0cde2d45d6829b4bf28f918421444cce789/code/modules/surgery/blood_filter.dm#L47-L52
It seems like it's specialcased to the blood filter *item*, not its tool behaviour.
https://github.com/tgstation/tgstation/blob/66b4b0cde2d45d6829b4bf28f918421444cce789/code/modules/surgery/blood_filter.dm#L64-L69
And even if we were to just change that value, its logic uses the blood filter item specifically, and so that wouldn't Just Work.

**Comment by Singul0** [comment]
specialcased items and it's consequences has been a disaster to the human race

**Comment by 00-Steven** [comment]
oops! yeah I totally forgot about this issue already honestly.

Buuuuut the welding tool one should be pretty doable, just a bit jank. regardless it's gonna be a bit jank.
I'll probably be working on something else for a bit, but hey now ya know what's fucked

**Comment by Singul0** [comment]
probably can make it into some kind of component? dunno never played with components before

**Comment by 00-Steven** [comment]
you should try them, it's really fun :)
